bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/rest/api/automowerconnect/AutomowerConnectApi.java

```diff
@@ -0,0 +1,141 @@
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Allows access to the AutomowerConnectApi
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AutomowerConnectApi extends HusqvarnaApi {
+
+    public AutomowerConnectApi(HttpClient httpClient) {
+        super(httpClient);
+    }
+
+    @Override
+    public String getBaseUrl() {
+        return "https://api.amc.husqvarna.dev/v1";
+    }
+
+    public MowerListResult getMowers(String appKey, String token) throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers");
+        request.method(HttpMethod.GET);
+
+        request.header("Authorization-Provider", "husqvarna");
+        request.header("Authorization", "Bearer " + token);
+        request.header("X-Api-Key", appKey);
+        request.header("Content-Type", "application/vnd.api+json");
+
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new AutomowerCommunicationException(e);
+        }
+
+        return parseResponse(response, MowerListResult.class);
+    }
+
+    public MowerResult getMower(String appKey, String token, String mowerId) throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers/" + mowerId);
+        request.method(HttpMethod.GET);
+
+        request.header("Authorization-Provider", "husqvarna");
+        request.header("Authorization", "Bearer " + token);
+        request.header("X-Api-Key", appKey);
+        request.header("Content-Type", "application/vnd.api+json");
+
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new AutomowerCommunicationException(e);
+        }
+
+        return parseResponse(response, MowerResult.class);
+    }
+
+    public void sendCommand(String appKey, String token, String id, MowerCommandRequest command)
+            throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/mowers/" + id + "/actions");
+        request.method(HttpMethod.POST);
+
+        request.header("Authorization-Provider", "husqvarna");
+        request.header("Authorization", "Bearer " + token);
+        request.header("X-Api-Key", appKey);
+        request.header("Content-Type", "application/vnd.api+json");
+
+        request.content(new StringContentProvider(gson.toJson(command)));
+
+        ContentResponse response;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new AutomowerCommunicationException(e);
+        }
+
+        checkForError(response, response.getStatus());
```

Comment:
You can put this in a separate method (except for the `content` call here) and then call that method from the 3 other methods to make the code simpler.

Reply:
yes, makes sense